manifests/osfamily/solaris.pp

```diff
@@ -32,6 +32,57 @@
         source => "puppet:///pe_packages/${pe_server_version}/${::platform_tag}/solaris-noask",
       }
     }
+    '11': {
+      class { 'puppet_agent::prepare::package':
+        package_file_name => $package_file_name,
+      }
+      contain puppet_agent::prepare::package
+
+      # Backup user configuration because solaris 11 will blow away
+      # /etc/puppetlabs/ when uninstalling the pe-* modules.
+      file { '/tmp/puppet_agent/':
+        ensure => directory,
+      }
+      exec { 'puppet_agent backup /etc/puppetlabs/':
+        command => 'cp -r /etc/puppetlabs/ /tmp/puppet_agent/',
+        require => [Exec['puppet_agent copy packages'], File['/tmp/puppet_agent/']],
+        path    => '/bin:/usr/bin:/sbin:/usr/sbin',
+      }
+
+      $pkgrepo_dir = '/etc/puppetlabs/installer/solaris.repo'
+
+      exec { 'puppet_agent remove existing repo':
+        command   => "pkgrepo remove -s '${pkgrepo_dir}' '*'",
```

Comment:
That is a good question.

I'm not sure where this step should fall with regard to package uninstallation. I basically copied these out of pe_repo to follow the same logic. Since pe_repo doesn't uninstall the old packages i just put this where I thought it would make the most sense.

I could try an agent upgrade with this step happening before the uninstall and see what that does. I was thinking specifically before the `/etc/puppetlabs/` backup since that directory is currently clobbered on package removal.